build-in-container/scripts/setup.sh

```diff
@@ -59,6 +59,37 @@ check_specs() {
     fi
 }
 
+# enable custom-repo.repo to install RPMs from
+setup_custom_repofile() {
+    echo "------------ Setting up custom repofile ------------"
+    echo -e "\n" >> $MARINER_BASE_DIR/toolkit/resources/manifests/package/local.repo
+    cat $RPM_repo_file >> $MARINER_BASE_DIR/toolkit/resources/manifests/package/local.repo
+    echo -e "\n" >> $MARINER_BASE_DIR/toolkit/resources/manifests/package/local.repo
+}
+
+# enable custom blob storage to install RPMs from
+setup_custom_repo_storage() {
+    echo "------------ Downloading RPMs from custom RPM blob storage container ------------"
+    #install azcopy
+    wget -O azcopy_v10.tar.gz https://aka.ms/downloadazcopy-v10-linux  || { echo "ERROR: Could not install azcopy"; exit 1; }
+    tar -xf azcopy_v10.tar.gz --strip-components=1
+    mv azcopy /bin/
+    rm -rf azcopy* NOTICE.txt
+
+    for i in $(echo $RPM_container_URL | tr "," "\n")
+    do
+        #download all RPMs from Azure $RPM_container_URL to $MARINER_BASE_DIR/build/rpm_cache/cache
+        azcopy copy $RPM_container_URL/* $MARINER_BASE_DIR/build/rpm_cache/cache
```

Comment:
Not exactly the same, but I'm working on a tool that supports this flow for actual repos (https://github.com/microsoft/CBL-Mariner/pull/5926).